universal-application-tool-0.0.1/app/views/LanguageUtils.java

```diff
@@ -0,0 +1,124 @@
+package views;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static j2html.TagCreator.div;
+import static j2html.TagCreator.input;
+import static j2html.TagCreator.label;
+import static j2html.TagCreator.option;
+import static j2html.TagCreator.select;
+
+import com.google.common.collect.ImmutableList;
+import j2html.attributes.Attr;
+import j2html.tags.ContainerTag;
+import j2html.tags.Tag;
+import java.util.Locale;
+import java.util.Optional;
+import javax.inject.Inject;
+import play.i18n.Lang;
+import play.i18n.Langs;
+import play.mvc.Http;
+import views.style.BaseStyles;
+import views.style.ReferenceClasses;
+import views.style.StyleUtils;
+import views.style.Styles;
+
+public class LanguageUtils {
+
+  public final ImmutableList<Locale> supportedLanguages;
+
+  @Inject
+  public LanguageUtils(Langs langs) {
+    this.supportedLanguages =
+        langs.availables().stream().map(Lang::toLocale).collect(toImmutableList());
+  }
+
+  public Optional<String> getPreferredLangage(Http.Request request) {
+    Optional<Http.Cookie> language = request.cookies().get("PLAY_LANG");
+    if (language.isPresent()) {
+      return Optional.of(language.get().value());
+    }
+    return Optional.empty();
+  }
```

Comment:
suggestion (blocking): Prefer the following to keep consistent with Play:

```java
// Inject MessagesApi in constructor, then:

Messages messages = messagesApi.preferred(request);
String locale = messages.lang().toString;
```

Need to finish  reading the rest, but using `Lang` or `Locale` seems safer than String - is there a strong reason to use String over a language type?

Reply:
Everything downstream requires a String (selecting a specific value in a dropdown/radio list, changing the lang html attribute) but I think it's probably better to let those use cases handle it with `Lang.toString()`.

Reply:
Agreed, this looks nice!